### PR TITLE
Set smaller threshold in internal edges of WaypointEdge.

### PR DESCRIPTION
**Pull request by florent-lamiraux**
This is useful for LevelSetEdges.

### CODE REVIEW

File: src/graph/edge.cc [hunk 1]
```diff
@@ -445,6 +447,22 @@ namespace hpp {
...
       }
 
+      void WaypointEdge::initialize ()
+      {
+        Edge::initialize();
+        // Set error threshold of internal edge to error threshold of
+        // waypoint edge divided by number of edges.
+        assert(targetConstraint()->configProjector());
+        value_type eps(targetConstraint()->configProjector()
+                       ->errorThreshold()/(value_type)edges_.size());
```
Comment (jmirabel) [inline src/graph/edge.cc:457]: What is the logic behind this value ?

File: src/graph/graph-component.cc [hunk 2]
```diff
@@ -140,7 +140,10 @@ namespace hpp {
 
       void GraphComponent::throwIfNotInitialized () const
       {
-        if (!isInit_ || (graph_.lock() && !graph_.lock()->isInit_)) throw std::logic_error ("The graph should have been initialized first.");
+        if (!isInit_){
```
Comment (jmirabel) [inline src/graph/graph-component.cc:143]: I think you need to check the graph isInit_ value. Why did you remove it ?

Comment (florent-lamiraux) [reply src/graph/graph-component.cc:143]: It was necessary. To modifiy the projector of an edge, the edge needs to be initialized. As I modify the projector of previously initialized edges in the initialization of the waypoint edge, the graph is not initialzed yet.

Comment (jmirabel) [reply src/graph/graph-component.cc:143]: The logic behind the isInit_ would be broken by this change because the constraints added to the graph are used in all the other components.

You could add a virtual method `GraphComponent::invalidate` that will set `isInit_ = false`. Then, reimplement it `Graph::invalidate` that calls `invalidate` on each GraphComponent of the graph (except itself, obviously). Then, in all places where you find `isInit_ = false`, call `invalidate` instead.

**Comment by florent-lamiraux** [comment]
@jmirabel: I have taken into account your comments.
I have also added a test to check that adding a constraint to the graph makes edges uninitialized.

File: src/graph/edge.cc [hunk 3]
```diff
@@ -526,6 +544,11 @@ namespace hpp {
 
       void WaypointEdge::nbWaypoints (const size_type number)
       {
+        if (edges_.size() > 1) {
+          HPP_THROW(std::runtime_error,
+                    "It is not allowed to resize a waypoint edge more than "
+                    "once: nb edges:" << edges_.size());
+        }
```
Comment (jmirabel) [inline src/graph/edge.cc:551]: Why is this not allowed ? Calling invalidate should be sufficient, no ?

File: CMakeLists.txt [hunk 4]
```diff
@@ -53,8 +51,8 @@ ADD_REQUIRED_DEPENDENCY("hpp-statistics >= 1.1")
 IF (HPP_MANIPULATION_HAS_WHOLEBODY_STEP)
    ADD_REQUIRED_DEPENDENCY("hpp-wholebody-step >= 4")
 ENDIF ()
-IF (TEST_UR5)
-  ADD_REQUIRED_DEPENDENCY("hpp_universal_robot")
+IF (BUILD_TESTING)
+  ADD_REQUIRED_DEPENDENCY("hpp-universal-robot")
```
Comment (jmirabel) [inline CMakeLists.txt:55]: Isn't it example-robot-data instead ?

Comment (florent-lamiraux) [reply CMakeLists.txt:55]: Yes it is. I will fix it.

File: src/graph/edge.cc (hunk 3 above)
Comment (florent-lamiraux) [reply src/graph/edge.cc:551]: I do not remember why I added this test. You are probably right, calling invalidate should be sufficient.
